Sources/Container.swift

```diff
@@ -58,9 +71,10 @@ public final class Container {
         parent: Container? = nil,
         defaultObjectScope: ObjectScope = .graph,
         behaviors: [Behavior] = [],
-        registeringClosure: (Container) -> Void = { _ in }
+        registeringClosure: (Container) -> Void = { _ in },
+        resolverHierarchyAccess: ResolverHierarchyAccess = .receiver
```

Comment:
I left the default here as the current behavior, but it would be nice to discuss changing the default in a future version.